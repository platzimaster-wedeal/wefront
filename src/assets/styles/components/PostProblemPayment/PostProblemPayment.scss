@import '../../partials/sizes.scss';
@import '../../partials/colors.scss';
@import '../../partials/fonts.scss';
@import '../../Breakpoints.scss';

.post-problem-payment {
  position: relative;
  min-width: 330px;
  width: 60%;
  display: flex;
  flex-flow: column nowrap;


  & label {
    font-family: $font-secondary-medium;
    font-weight: 500;
    color: $color-black-90;
    padding: $space-size-extra-small 0;
    @include font-size($font-size-normal);
  }

  &__action-price {
    @include font-size($font-size-normal);

    & span {
      color: $color-primary;
      font-family: $font-secondary-medium;
    }
  }

  &__range-price {
    display: inline-flex;
    width: 100%;
    justify-content: space-between;
    padding: $space-size-extra-small $space-size-normal;
    & small {
      @include font-size($font-size-small);

      @include media(tablet) {
        @include font-size($font-size-normal);
      }
    } 
  }

  & button {
    margin-right: $space-size-normal;
  }
}
