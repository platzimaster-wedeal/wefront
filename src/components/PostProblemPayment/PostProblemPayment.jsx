import React, {useEffect} from "react";

// Styles
import "../../assets/styles/components/PostProblemPayment/PostProblemPayment.scss";

// Components
import Button from "../Buttons/Button";

const PostProblemPayment = ({ agreed = false, agreedPrice = 0, salaryMin = 0, salaryMax = 1000, setState, state }) => {

 return (
  <div className="post-problem-payment">
   <label className="post-problem-payment__title" htmlFor="problem-price">
    Payment
   </label>

   <p className="post-problem-payment__action-price">
    {agreed != true ? "Select your price:" : "Agreed Price:"}{" "}
    <span className="post-problem-payment__current-price">
     ${Number(agreedPrice) > 0 ? Number(agreedPrice) : state}
    </span>
   </p>
   {agreed != true && (
    <>
     <div className="post-problem-payment__range-price">
      <small>
       min: <span> ${salaryMin} </span>
      </small>
      <small>
       max: <span> ${salaryMax} </span>
      </small>
     </div>

     <input type="range" onChange={setState} max={salaryMax} min={salaryMin} />
    </>
   )}
  </div>
 );
};

export default PostProblemPayment;
