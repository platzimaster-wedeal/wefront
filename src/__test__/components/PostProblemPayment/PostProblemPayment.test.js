import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import PostProblemPayment from "../../../components/PostProblemPayment/PostProblemPayment";

describe("Test for the component PostProblemPayment :D", () => {
 const PostProblemPaymentMounted = mount(<PostProblemPayment />);
 const PostProblemPaymentCreated = create(<PostProblemPayment />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(PostProblemPaymentCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(PostProblemPaymentMounted.length).toEqual(1);
  });
  test("Should the label of the problem be a string", () => {
   expect(
    typeof PostProblemPaymentMounted.find(".post-problem-payment__title").text()
   ).toEqual("string");
  });
  test("Should the price of the problem be a number", () => {
   expect(
    typeof Number(
     PostProblemPaymentMounted.find(".post-problem-payment__current-price")
      .text()
      .split("$")[1]
    )
   ).toEqual("number");
  });
 });
});
