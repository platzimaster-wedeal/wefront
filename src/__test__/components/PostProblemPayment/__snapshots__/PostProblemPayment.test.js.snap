// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component PostProblemPayment :D Visual tests for the component should the comopnent have the same tree 1`] = `
<div
  className="post-problem-payment"
>
  <label
    className="post-problem-payment__title"
    htmlFor="problem-price"
  >
    Payment
  </label>
  <p
    className="post-problem-payment__action-price"
  >
    Select your price:
     
    <span
      className="post-problem-payment__current-price"
    >
      $
      0
    </span>
  </p>
  <div
    className="post-problem-payment__range-price"
  >
    <small>
      min: 
      <span>
         $0 
      </span>
    </small>
    <small>
      max: 
      <span>
         $1000 
      </span>
    </small>
  </div>
  <input
    max={1000}
    min={0}
    onChange={[Function]}
    type="range"
  />
</div>
`;
